python/src/worker/tasks/factory.py

```diff
@@ -36,39 +36,57 @@ class TaskFactory:
             GetDoubletScore,
             GetMitochondrialContent,
             MarkerHeatmap,
-            GetExpressionCellSets
+            GetExpressionCellSets,
         )
     }
 
     def __init__(self):
         self.count_matrix = CountMatrix()
         self.count_matrix.sync()
 
+    def _log_exception(self, task, error_object):
+        trace = (
+            f"Exception for task {task.__class__.__name__}:\n"
+            f"{traceback.format_exc()}"
+        )
+
+        error(trace)
+        xray_recorder.current_segment().add_exception(error_object, trace)
+
+        return trace
+
     def submit(self, msg):
         task = self._factory(msg)
 
         # Try to perform task. If fails, send back an error to the API.
         try:
             result = task.compute()
             return result
-        except Exception as e:
-            trace = f"Exception for task {task.__class__.__name__}:\n" \
-                    f"{traceback.format_exc()}"
 
-            error(trace)
-            xray_recorder.current_segment().add_exception(e, trace)
+        except Exception as e:
+            trace = self._log_exception(task, e)
 
-            # Only send real traces in development.
             if config.CLUSTER_ENV == "development":
                 result = Result(trace, error=True)
```

Comment:
this would never be returned the way you re-structured the code

Reply:
Do we still need this? In development, we can see the stack trace in the worker terminal window, and we don't need to send the trace to the UI in staging or production. I think we can remove this?

Reply:
I just saw your comment below. I agree that we do not need this anymore, as we can easily see errors in development, so I'm removing it.